### PR TITLE
[front] fix: more improvements in vaults api

### PR DESCRIPTION
**Pull request by tdraier**
## Description

Follow up for https://github.com/dust-tt/dust/pull/6713

## Risk

None

## Deploy Plan

deploy `front`

### CODE REVIEW

File: front/lib/api/agent_data_sources.ts [hunk 1]
```diff
@@ -128,3 +130,53 @@ export async function getDataSourceUsage({
...
+  auth: Authenticator;
+  dataSourceView: DataSourceViewType;
+}): Promise<number> {
+  const owner = auth.workspace();
+
+  // This condition is critical it checks that we can identify the workspace and that the current
+  // auth is a user for this workspace. Checking `auth.isUser()` is critical as it would otherwise
+  // be possible to access data sources without being authenticated.
+  if (!owner || !auth.isUser()) {
+    return 0;
```
Comment (flvndvd) [inline front/lib/api/agent_data_sources.ts:147]: Shouldn't this return an error or null instead?

File: front/lib/api/agent_data_sources.ts [hunk 2]
```diff
@@ -128,3 +130,53 @@ export async function getDataSourceUsage({
...
+        model: DataSourceViewModel,
+        as: "dataSourceView",
+        where: {
+          workspaceId: owner.id,
+        },
+        attributes: [],
+      },
+      {
+        model: AgentRetrievalConfiguration,
+        as: "agent_retrieval_configuration",
```
Comment (flvndvd) [inline front/lib/api/agent_data_sources.ts:165]: can we stick to camel case?

File: front/lib/api/vaults.ts [hunk 3]
```diff
@@ -34,16 +34,18 @@ export const getDataSourceInfos = async (
 ): Promise<DataSourceOrViewInfo[]> => {
   const dataSources = await DataSourceResource.listByVault(auth, vault);
 
-  return dataSources.map((dataSource) => getDataSourceInfo(dataSource));
+  return Promise.all(
+    dataSources.map((dataSource) => getDataSourceInfo(auth, dataSource))
```
Comment (flvndvd) [inline front/lib/api/vaults.ts:38]: How efficient is the usage query?

File: front/lib/resources/data_source_view_resource.ts [hunk 4]
```diff
@@ -265,14 +258,24 @@ export class DataSourceViewResource extends ResourceWithVault<DataSourceViewMode
...
+      sId: this.sId,
       createdAt: this.createdAt.getTime(),
       parentsIn: this.parentsIn,
-      sId: this.sId,
       updatedAt: this.updatedAt.getTime(),
+      connectorId: this.dataSource.connectorId,
+      connectorProvider: this.dataSource.connectorProvider,
+      name: this.dataSource.name,
+      description: this.dataSource.description,
+      dustAPIProjectId: this.dataSource.dustAPIProjectId,
```
Comment (flvndvd) [inline front/lib/resources/data_source_view_resource.ts:278]: Why do we need all those fields?

File: front/lib/resources/vault_resource.ts [hunk 5]
```diff
@@ -159,18 +163,22 @@ export class VaultResource extends BaseResource<VaultModel> {
...
         workspaceId: owner.id,
       },
     });
 
-    if (!vault) {
+    if (!vaultModel) {
+      return null;
+    }
+    const vault = new this(VaultModel, vaultModel.get());
+    if (!auth.isAdmin() && !auth.hasPermission([vault.acl()], "read")) {
```
Comment (flvndvd) [inline front/lib/resources/vault_resource.ts:177]: I thought we agreed on adding this permission layer only on the list?

File: front/lib/api/agent_data_sources.ts (hunk 2 above)
Comment (tdraier) [reply front/lib/api/agent_data_sources.ts:165]: this is how it's defined in https://github.com/dust-tt/dust/blob/3f69681058786326460d008173eff0f9002900ed/front/lib/models/assistant/actions/retrieval.ts#L95  (and so in all other methods in agent_data_sources)

File: front/lib/resources/vault_resource.ts (hunk 5 above)
Comment (tdraier) [reply front/lib/resources/vault_resource.ts:177]: right .. forgot to remove that one

File: front/lib/api/vaults.ts (hunk 3 above)
Comment (tdraier) [reply front/lib/api/vaults.ts:39]: can be improved : 

```
explain SELECT "agent_data_source_configuration"."id" AS "count" FROM "agent_data_source_configurations" AS "agent_data_source_configuration" INNER JOIN "data_source_views" AS "dataSourceView" ON "agent_data_source_configuration"."dataSourceViewId" = "dataSourceView"."id" AND "dataSourceView"."workspaceId" = 1 INNER JOIN "agent_retrieval_configurations" AS "agent_retrieval_configuration" ON "agent_data_source_configuration"."retrievalConfigurationId" = "agent_retrieval_configuration"."id" INNER JOIN "agent_configurations" AS "agent_retrieval_configuration->agent_configuration" ON "agent_retrieval_configuration"."agentConfigurationId" = "agent_retrieval_configuration->agent_configuration"."id" AND "agent_retrieval_configuration->agent_configuration"."status" = 'active' WHERE "agent_data_source_configuration"."dataSourceViewId" = 10;
                                                                                           QUERY PLAN
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Nested Loop  (cost=6.61..20.43 rows=1 width=4)
   ->  Nested Loop  (cost=6.61..19.35 rows=1 width=8)
         ->  Hash Join  (cost=6.46..16.73 rows=1 width=4)
               Hash Cond: (agent_retrieval_configuration."agentConfigurationId" = "agent_retrieval_configuration->agent_configuration".id)
               ->  Seq Scan on agent_retrieval_configurations agent_retrieval_configuration  (cost=0.00..10.20 rows=20 width=8)
               ->  Hash  (cost=6.44..6.44 rows=2 width=4)
                     ->  Seq Scan on agent_configurations "agent_retrieval_configuration->agent_configuration"  (cost=0.00..6.44 rows=2 width=4)
                           Filter: ((status)::text = 'active'::text)
         ->  Index Scan using agent_data_source_configurations_retrieval_configuration_id on agent_data_source_configurations agent_data_source_configuration  (cost=0.15..2.61 rows=1 width=12)
               Index Cond: ("retrievalConfigurationId" = agent_retrieval_configuration.id)
               Filter: ("dataSourceViewId" = 10)
   ->  Seq Scan on data_source_views "dataSourceView"  (cost=0.00..1.07 rows=1 width=4)
         Filter: ((id = 10) AND ("workspaceId" = 1))
(13 rows)
```

I also notice that the query we have is not checking the process_actions , it should be updated - we may have to do 2 separate queries .. ?

File: front/lib/api/agent_data_sources.ts (hunk 2 above)
Comment (flvndvd) [reply front/lib/api/agent_data_sources.ts]: Arf 🙄

File: front/lib/api/vaults.ts (hunk 3 above)
Comment (flvndvd) [reply front/lib/api/vaults.ts:39]: Indeed. Playing a bit more with it, I feel like those screens are pretty slow to load already. Should we consider caching?

File: front/lib/resources/data_source_view_resource.ts (hunk 4 above)
Comment (tdraier) [reply front/lib/resources/data_source_view_resource.ts:278]: it's to support DataSourceOrViewInfo and have consistent data between datasources and views. But it's not really needed in the datasourceview json directly, and we could keep the logic of merging data in getDataSourceViewInfo.

File: front/lib/api/vaults.ts (hunk 3 above)
Comment (tdraier) [reply front/lib/api/vaults.ts:39]: created a task here : https://github.com/dust-tt/tasks/issues/1152

File: front/pages/api/w/[wId]/vaults/index.ts [hunk 6]
```diff
@@ -43,12 +43,7 @@ async function handler(
       const vaults = await VaultResource.listWorkspaceVaults(auth);
 
       return res.status(200).json({
-        vaults: vaults
-          .filter(
-            (vault) =>
-              auth.isAdmin() || auth.hasPermission([vault.acl()], "read")
-          )
-          .map((vault) => vault.toJSON()),
+        vaults: vaults.map((vault) => vault.toJSON()),
```
Comment (JulesBelveze) [inline front/pages/api/w/[wId]/vaults/index.ts:46]: ooc: any reason we are removing the permission filter?

**Review by JulesBelveze — APPROVED** [review]
LGTM, I'm afraid the DB is gonna get hammered though..

File: front/pages/api/w/[wId]/vaults/index.ts (hunk 6 above)
Comment (tdraier) [reply front/pages/api/w/[wId]/vaults/index.ts:46]: it;s moved in the resource

**Pull request merged**